### DISCUSSION TITLE
Building the timeline is slow

### DISCUSSION THREAD

**Discussion by jagjordi** [immich-app/immich#4029] [Q&A]
Hello,

Is there any reason why everytime I open the app I see the "Building the timeline" for a significant amount of time ~0.5 to 1 minute? Is it something wrong with my phone, or is it supposed to be like that. In the browser its very fast (I have arround 75k photos and 10k videos)

**Comment by alextran1502** [comment]
This issue will be fixed when this PR is merged https://github.com/immich-app/immich/pull/3945

**Reply by jagjordi** [reply]
That is great! Any estimate on when it will be released?

**Reply by alextran1502** [reply]
I will plan to push out a release before the end of the week. I am however on vacation so I don't have access to my phone number where Apple will send in the 2FA code so iOS release will be later

**Reply by jagjordi** [reply]
Good that I use Android then ;)

**Comment by MozPri** [comment]
There is no notable change after the update to 1.78 (Android). I still get "Building the timeline" every time I start the app for about 10 seconds, the same as before. I have about 32k assets.

**Reply by alextran1502** [reply]
What is your server version?

**Reply by jagjordi** [reply]
its the same for me, no noticeable improvement. Both server and app v1.78.0

**Reply by MozPri** [reply]
Version 1.78.0, both app and server.

**Reply by jagjordi** [reply]
[Immich_log_2023-09-13T180613.182912.csv](https://github.com/immich-app/immich/files/12606307/Immich_log_2023-09-13T180613.182912.csv)

**Reply by mosalski** [reply]
its the same for me, takes about 10 seconds on every app start, but according to the log it loads in less than in a second. May it be related with UI loader animation?

**Comment by jagjordi** [comment]
Any update regarding this @alextran1502  ? Do you need additional debugging data?

**Reply by alextran1502** [reply]
We have a PR opened to address this, I haven't had a chance to get to it yet

**Comment by stodge** [comment]
Is it supposed to take many minutes to build the timeline the first time I run the iOS app? Currently it's still building after 5 minutes.

**Reply by alextran1502** [reply]
@stodge no it shouldn't. 

What is your app version?

Is there any assets on your instance already?

**Reply by stodge** [reply]
Server is 1.87.0 and the iOS app is 1.87.0 build 127. No there were no assets.

**Reply by alextran1502** [reply]
So you cannot expect the timeline to show anything if there is no assets

**Reply by stodge** [reply]
I know that. It displayed the message "Building Timeline" until I uploaded something, implying it was busy doing something. I think it makes more sense to show an empty timeline, or "No Photos", rather than "Building Timeline" with an animated icon?
